### PR TITLE
Remove HomogeneousTupleVariable

### LINKED ISSUE #1582: Deprecate HomogeneousTupleVariable
A tuple does not need to have its own type of Variable. This distinction makes it impossible to create tuples in classes. Homogeneous tuples can be handled easily using class definitions.

### PR DESCRIPTION
**Pull request by EmilyBourne**
Remove the class `HomogeneousTupleVariable`. This is no longer needed now that we have `TypedAstNode.class_type`. Removing this class allows us to store homogeneous tuples in classes. Fixes #1582 .

### CODE REVIEW

File: pyccel/parser/semantic.py [hunk 1]
```diff
@@ -3043,7 +2980,7 @@ def _visit_Assign(self, expr):
                     new_lhs.append( self._assign_lhs_variable(l, d, r, new_expressions, isinstance(expr, AugAssign),arr_in_multirets=r.rank>0 ) )
                 lhs = PythonTuple(*new_lhs)
 
-            elif isinstance(rhs, HomogeneousTupleVariable):
+            elif isinstance(rhs.class_type, NativeHomogeneousTuple):
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/parser/semantic.py:2983]: /bot accept
Known issue #1392

**Comment by EmilyBourne** [comment]
Blocked by #1590

**Review by yguclu — APPROVED** [review]

**Pull request merged**